### DISCUSSION TITLE
Using mermaid to document digital logic

### DISCUSSION THREAD

**Discussion by arquer** [mermaid-js/mermaid#3991] [Q&A]
Hi, I am an FPGA designer and I am very interested in using mermaid to document digital logic designs.
Looking through the different diagram/graphs/charts/... that mermaid offers I have not seen anything quiet like what I am looking for. Ideally I would like mermaid to allow me to draw something like this:

![image](https://user-images.githubusercontent.com/10212172/212464266-cd2b14eb-c972-4e8d-9137-8b23d4180ef5.png)

Maybe basic gates too:

![image](https://user-images.githubusercontent.com/10212172/212464343-cb99f884-fc9b-4415-8870-ded84c2b6241.png)

My idea would be to have some basic/low-level building blocks like flip-flops, LUTs, Multiplexers, Demultiplexers. 
Is mermaid capable of something like this currently?

**Comment by taaaf11** [comment]
Any updates?

**Comment by TNTY100** [comment]
I would love to see this happen. It would really bring the tool to another level. 

Edit : I'll add this to an Idea.

Edit 2 : Idea in question #7347
